primary/src/header_waiter.rs

```diff
@@ -231,7 +231,7 @@ impl HeaderWaiter {
                                     .name;
 
                                 // TODO [issue #423]: This network transmission needs to be reliable: the worker may crash-recover.
-                                let message = PrimaryWorkerMessage::Synchronize(digests, author.clone());
+                                let message = WorkerSynchronizeMessage{digests, target: author.clone()};
```

Comment:
Lets make a constructor function for `WorkerSynchronizeMessage` and use it here to give us flexibility.

Reply:
I think we should have the same convention for all of the RPC request/response messages. Right now none of them have constructors. I don't have a strong opinion either way but maybe worth a discussion and then we could change them all together in a separate PR if desired. @gdanezis what do you think the benefits are? @bmwill thoughts?

Reply:
No strong feelings, just a suggestion. In my experience its harder to refactor when we init structures directly, but its a case by case thing.